static/src/javascripts/projects/common/modules/commercial/adfree/maintain-adfree-freshness.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'common/utils/cookies',
+    'common/utils/config',
+    'common/utils/storage',
+    'common/modules/identity/api',
+    'common/modules/commercial/adfree/renew-adfree-status'
+], function (
+    cookies,
+    config,
+    storage,
+    identity,
+    renewAdfreeStatus
+) {
+    return maintainAdfreeFreshness;
+
+    function maintainAdfreeFreshness() {
+        if (featureEnabled() && identity.isUserLoggedIn() && userNeedsNewAdfreeCookie()) {
+            renewAdfreeStatus.renew();
+        }
+    }
+
+    function userNeedsNewAdfreeCookie() {
+        var adfreeCookie = cookies.get('gu_adfree_user');
+        return (adfreeCookie === null) || adfreeCookieStale();
+    }
+
+    function adfreeCookieStale() {
+        var adfreeCookieExpiry = storage.local.get('gu.adfree.user.expiry'),
```

Comment:
As I remember it correctly, cookies are very grey area. EU law + users don't like them so we are generally trying to avoid them as much as possible.

However I agree that it would be nicer to have everything at one place.